File: src/main/java/com/frank/entity/courier/Courier.java
package com.frank.entity.courier;

import com.frank.entity.order.Order;

import java.time.LocalDate;
import java.util.List;
import java.util.Queue;
import java.util.concurrent.ConcurrentLinkedDeque;

public class Courier {

    final private Queue<Order> orderQueue = new ConcurrentLinkedDeque<>();

    private List<LocalDate> availability; // TODO populate this field someway

    public void receive(Order order) {
        order.setCourier(this);
    }

    public List<LocalDate> getAvailability() {
        return availability;
    }

}

File: src/main/kotlin/com/frank/entity/courier/Courier.kt
package com.frank.entity.courier

import com.frank.entity.order.Order

import java.time.LocalDate
import java.util.Queue
import java.util.concurrent.ConcurrentLinkedDeque

class Courier {

    private val orderQueue = ConcurrentLinkedDeque<Order>()

    val availability: List<LocalDate>? = null // TODO populate this field someway

    fun receive(order: Order) {
        order.courier = this
    }

}
